### PR TITLE
Go 1.21

### PR DESCRIPTION
**Pull request by roman-khimov**
@AnnaShaleva, @AliceInHunterland: please review it carefully, it's too easy to make a mistake with this amount of repetitive changes.

Go policy change is scheduled to #3310, I'd like to have 1.23 now at least because of `strings.Compare` that we use already in code.

### CODE REVIEW

**Comment by roman-khimov** [comment]
cmp.Or has leaked there, to be fixed.

**Comment by roman-khimov** [comment]
`slices.Concat` to be moved out as well.

File: pkg/services/oracle/nodes.go
Comment (AnnaShaleva) [inline pkg/services/oracle/nodes.go:19]: The same should be implemented for Notary, I'm not sure why we don't have this check for Notary service.

File: pkg/smartcontract/manifest/parameter.go
Comment (AnnaShaleva) [inline pkg/smartcontract/manifest/parameter.go]: This cycle is not needed, there are only two elements. Just use `return cmp(x[0], x[1])`.

File: pkg/network/server.go
Comment (AnnaShaleva) [inline pkg/network/server.go:1451]: Something isn't clear here. Why do we need `peersNumber` variable then if it matches `len(heights)`?

File: cli/query/query.go
Comment (AnnaShaleva) [inline cli/query/query.go:202]: How about votes comparison? This code was just removed.

File: pkg/core/interop/context.go
Comment (AnnaShaleva) [inline pkg/core/interop/context.go:398]: Is this search behaviour deterministic from run to run? It should be, but we'd better be 100% sure. Consider the following case: malicious contract is deployed, it has two `verify` methods that differ by parameters count. We need the result of this search be deterministic from run to run and also matching C# node.

File: pkg/core/mpt/trie_store.go
Comment (AnnaShaleva) [inline pkg/core/mpt/trie_store.go:40]: It's a good change but it seems to me that we have a lot of other cases of "unsupported" usage. Should we consider replacing most of them?

**Review by AnnaShaleva — COMMENTED** [review]
Sorry, but I can't approve this PR without mainnet/testnet states comparison.

File: pkg/util/uint256.go
Comment (roman-khimov) [reply pkg/util/uint256.go:61]: It'd incur more allocations, we need some place to reverse bytes (`b` can't be changed), that's likely `u`, but then we shouldn't try reversing any input (it can be of a wrong length), so there you have a length check above, now when you've done all of that --- why reusing `Uint256DecodeBytesBE`?

File: pkg/smartcontract/manifest/parameter.go
Comment (roman-khimov) [reply pkg/smartcontract/manifest/parameter.go]: It can have more of them.

File: pkg/network/server.go
Comment (roman-khimov) [reply pkg/network/server.go:1451]: Fair enough.

File: pkg/smartcontract/manifest/parameter.go
Comment (AnnaShaleva) [reply pkg/smartcontract/manifest/parameter.go]: OK, I missed there's no `return` after sorting code above.

File: pkg/core/mpt/trie_store.go
Comment (roman-khimov) [reply pkg/core/mpt/trie_store.go:40]: Maybe. It's more of a question of how to find them.

File: pkg/core/interop/context.go
Comment (roman-khimov) [reply pkg/core/interop/context.go:398]: I've restored the original behavior, but looks like C# does `Methods.FirstOrDefault` from the Manifest which is not sorted at all in general. But `HFSpecificContractMD` is only used for native contracts, they don't have a lot of surprises.

File: pkg/services/oracle/nodes.go
Comment (roman-khimov) [reply pkg/services/oracle/nodes.go:19]: Because of m/n vs 1/n? There is a check for the key presence.

**Comment by roman-khimov** [comment]
A bit more of `ErrUnsupported` added.

**Review by AnnaShaleva — COMMENTED** [review]
LGTM, but I'd still prefer to check states before the merge.

File: pkg/core/interop/context.go
Comment (AnnaShaleva) [reply pkg/core/interop/context.go]: You're right, I missed that it's for natives only.

**Comment by roman-khimov** [comment]
Testnet OK up to 4543634.

**Review by AnnaShaleva — APPROVED** [review]

**Comment by roman-khimov** [comment]
Mainnet OK up to 5928863.

**Review by AliceInHunterland — APPROVED** [review]

**Pull request merged**
